packages/m365_defender/data_stream/alert/elasticsearch/ingest_pipeline/default.yml

```
@@ -0,0 +1,2530 @@
+          target_field: _ingest._value.registry_value
+          ignore_missing: true
+  - foreach:
+      field: json.evidence
+      tag: foreach_evidence_append_registryValue
+      if: ctx.json?.evidence instanceof List
+      processor:
+        append:
+          field: registry.value
+          tag: append_foreach_evidence_registryValue
+          value: '{{{_ingest._value.registry_value}}}'
+          allow_duplicates: false
+  - foreach:
+      field: json.evidence
+      tag: foreach_evidence_rename_registryValueName
+      if: ctx.json?.evidence instanceof List
+      processor:
+        rename:
+          field: _ingest._value.registryValueName
+          tag: rename_foreach_evidence_registryValueName
+          target_field: _ingest._value.registry_value_name
+          ignore_missing: true
+  - foreach:
+      field: json.evidence
+      tag: foreach_evidence_rename_securityGroupId
+      if: ctx.json?.evidence instanceof List
+      processor:
+        rename:
+          field: _ingest._value.securityGroupId
+          tag: rename_foreach_evidence_securityGroupId
+          target_field: _ingest._value.security_group_id
+          ignore_missing: true
+  - foreach:
+      field: json.evidence
+      tag: foreach_evidence_append_securityGroupId
+      if: ctx.json?.evidence instanceof List
+      processor:
+        append:
+          field: group.id
+          tag: append_foreach_evidence_securityGroupId
+          value: '{{{_ingest._value.security_group_id}}}'
+          allow_duplicates: false
+  - script:
+      description: Map ecs fields based on evidence type.
+      tag: script_to_map_evidence_ecs_fields
+      lang: painless
+      if: ctx.json?.evidence instanceof List
+      source:
+        def convertToOrderedArray(def list) {
+          def result = new ArrayList();
+          for (element in list) {
+            result.add(element);
+          }
+          Collections.sort(result);
+          return result;
+        }
+        if (!(ctx.cloud instanceof HashMap)) {
+          ctx.cloud = new HashMap();
+        }
+        def cloudProvider = new HashSet();
+        if (!(ctx.group instanceof HashMap)) {
+          ctx.group = new HashMap();
+        }
+        def groupName = new HashSet();
+        if (!(ctx.host instanceof HashMap)) {
+          ctx.host = new HashMap();
+        }
+        def hostId = new HashSet();
+        if (!(ctx.user instanceof HashMap)) {
+          ctx.user = new HashMap();
+        }
+        def userDomain = new HashSet();
+        def userId = new HashSet();
+        def userName = new HashSet();
+        def userEmail = new HashSet();
+        if (!(ctx.process instanceof HashMap)) {
+          ctx.process = new HashMap();
+        }
+        ctx.process.user = new HashMap();
+        def processUserId = new HashSet();
+        def processUserName = new HashSet();
+        for (evidence in ctx.json.evidence) {
+          if (evidence?.odata_type != null) {
```

Just for this first condition, can we invert the logic and continue instead. This will de-indent the rest of the loop body by one level.
```suggestion
          if (evidence?.odata_type == null) {
            continue
          }
```